src/include/txn.i

```diff
@@ -155,6 +148,13 @@ __wt_txn_visible(WT_SESSION_IMPL *session, uint64_t id)
 		return (1);
 
 	/*
+	 * Eviction only sees globally visible updates, or if there is a
+	 * checkpoint transaction running, use its transaction.
+	 */
+	if (txn->isolation == TXN_ISO_EVICTION)
+		return (__wt_txn_visible_all(session, id));
+
+	/*
```

Comment:
@michaelcahill, I think this change is safe because as far as I can tell, the result is the same as if we called `__wt_txn_visible_all` because  that code calls `TXNID_LT` on the passed-in transaction ID, which will always succeed for `WT_TXN_NONE` and always fail for `WT_TXN_ABORTED`.

I'm kind of confused why it's OK to always fail in eviction for `WT_TXN_ABORTED`?